packages/core/src/agent/Agent.ts

```diff
@@ -214,6 +214,28 @@ export class Agent {
     this._isInitialized = false
   }
 
+  public async rotateKey(walletConfig: WalletConfig) {
+    
+    // close wallet if still initialized
```

Comment:
Could you make sure your code passes validation using `yarn validate`?

Reply:
OK I agree, I moved the function to wallet interface then added an implementation in WalletModule and of course IndyWallet, however, we still need to access that from the agent side cuz you still need to shutdown the agent correctly like stop transports and you know all those buzz and whistles going on the "shutdown()" which is accessible on the Agent level. 

So the current implementation would be 

Access rekey from the Agent -> the Agent will make sure it shutdown properly -> execute the "rekey" method of the Wallet module
Is that correct? excuse me here am like a baby learning first steps :)

Reply:
No I'd say you need to shutdown the agent yourself.

So rather something like:

```ts
await agent.shutdown()
await agent.wallet.rotateKey()
await agent.initialize()
```

Reply:
Got it, so the shutting down will occur on the wallet side (controller)